apps/web/src/components/NotificationList/NotificationListPage.tsx

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { useQuery } from "urql";
+import { useSearchParams } from "react-router-dom";
+
+import { graphql } from "@gc-digital-talent/graphql";
+import { unpackMaybes } from "@gc-digital-talent/helpers";
+import { Link, Pending, Well } from "@gc-digital-talent/ui";
+
+import NotificationItem from "./NotificationItem";
+
+const Notifications_Query = graphql(/* GraphQL */ `
+  query Notifications(
+    $where: NotificationFilterInput
+    $first: Int
+    $page: Int
+  ) {
+    notifications(where: $where, first: $first, page: $page) {
+      paginatorInfo {
+        hasMorePages
+      }
+      data {
+        id
+        ...NotificationItem
+      }
+    }
+  }
+`);
+
+const PER_PAGE = 10;
+
+interface NotificationPageProps {
+  page: number;
+  onlyUnread?: boolean;
+  isLastPage?: boolean;
+}
+
+const NotificationListPage = ({
+  page,
+  onlyUnread,
+  isLastPage,
+}: NotificationPageProps) => {
+  const intl = useIntl();
+  const [searchParams] = useSearchParams();
+  searchParams.set("page", `${page + 1}`);
+  const [{ data, fetching, error }] = useQuery({
```

Comment:
Has there been any discussion about whether this page should be "live"?  Like, should this be polling for new notifications in the background?

Reply:
I think that is a good idea! Can we do that right now or is there something we need first to support it?

Reply:
If we want to do simple polling I guess you could just add a repeating timeout and rerun the query.  If we want to do graphql subscriptions or websockets or something we'll probably need to add some plumbing.  This is more important for the nav bar notification, I think.  Worth bringing up in standup.

Reply:
Yeah, thanks. I likely will. I'm trying to think how we could do this with `graphql` and pages 🤔 I guess we could have a general query that polls for the latest items since the last run and append them on and then set the offset for pagination to however many we loaded from that general polled version? 🤷‍♀️

Reply:
okay, it is kinda messy but I got this working in [fix markup, add polling to list](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9755/commits/89d6dcf36810d302d2103c70fee6f6f7e72bd6f7) but have it disabled for now. Not sure we want it on the main page but it is there for the dialog since I plan on using the same list component. If you want to test:

1. Change the `live` prop to `true` for the list on the page
2. Change the delay in seconds to something short
3. Login as `admin@test.com`
4. Open two tabs one on `/applicant/notifications` and the other on `/admin/users/{userId}` where `{userId}` is the ID of `admin@test.com`
5. Add the user to a pool on the admin page
6. Change their status
7. Observe the notifications page updates within the delay you set

Reply:
Cool, I've never used portals manually before!  I think there's some trouble with the polling logic though.  I get the refresh requests coming through in chains of two or three.  For example, I have the interval set to 10 seconds here:
![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/8978655/d04af046-ca0a-4e55-b016-b66c102d4ab3)  
Also, by only querying for "createdAt"  the live updating misses items that should be removed after delete and mark-as-read.  I'm pretty skeptical of trying to live update paged data.  How can you do an incremental update with incomplete data?  Unless we can query for some sort of changelog or pub-sub I think we need to accept that paged data is going to jump around when it is updated.

Reply:
Yeah, I'm not too sure what to do with this. What I have there is super hacky. `urql` expects you to be using the graphcache to get polling w/ pagination working.

I think it might be easier if we only use polling without the pagination until we have graphcache working :thinking: 

Or, get subscriptions working but that seems like alot more work.

Reply:
@tristan-orourke Does the main notifications page need to be live updated for now? It is technically disabled right now. We could update the dialog to be live updated and not paginated to keep this simple to start 🤔 

In regards the timeout running multiple times... I can't figure that out and am wondering if it is related to `useEffect` firing twice. Though, I did just use the suggestion that was mentioned when `urql` removed polling support.

https://github.com/urql-graphql/urql/pull/1374

Reply:
It seems like we're running into known issues with offset-based pagination 
https://www.sitepoint.com/paginating-real-time-data-cursor-based-pagination/#identifying-issues-in-real-time-data-pagination

Reply:
@esizer did you see this urql example of [infinite pagination](https://github.com/urql-graphql/urql/tree/main/examples/with-infinite-pagination)? It doesn't require graphecache. It uses recursive components, which is neat - I don't think I've ever seen that pattern before.

It does, however, use cursor-type pagination, which Lighthouse [doesn't really support yet](https://lighthouse-php.com/6/api-reference/directives.html#pagination-type).

Reply:
For the record, I don't care if live updates or "Load more" removes items which have been removed in another tab. I think needing to refresh the page to see that would not be surprising to a user.

Reply:
I'm more concerned about new notifications appearing properly, and not rendering duplicates because paging offsets have been messed up.

Reply:
Yeah, that was the reason I did not do that. We need cursor based pagination 😢 Though it is nice to know I was wrong about that style requiring graphcache! In their docs they only show it with graphcache 😅

Reply:
> Does the main notifications page need to be live updated for now? It is technically disabled right now. We could update the dialog to be live updated and not paginated to keep this simple to start 🤔

That's a great suggestion, let's go with that for now!

Reply:
okay, so until we can get cursor pagination to make it a smoother scenario for live notifications I have updated the component too disable pagination for the dialog but enabled it on this page while keeping it static.

[allow unpaginated notification list](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9755/commits/d29cf8f7e01142643033a3708c5b4a4cb4281f15)